lock-api/src/test/java/com/palantir/lock/client/MultiClientCommitTimestampGetterTest.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.Futures;
+import com.palantir.atlasdb.autobatch.BatchElement;
+import com.palantir.atlasdb.autobatch.DisruptorAutobatcher.DisruptorFuture;
+import com.palantir.atlasdb.timelock.api.GetCommitTimestampsRequest;
+import com.palantir.atlasdb.timelock.api.GetCommitTimestampsResponse;
+import com.palantir.atlasdb.timelock.api.Namespace;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.lock.client.MultiClientCommitTimestampGetter.NamespacedRequest;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.watch.LockWatchEventCache;
+import com.palantir.lock.watch.LockWatchStateUpdate;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import one.util.streamex.StreamEx;
+import org.junit.Test;
+
+public class MultiClientCommitTimestampGetterTest {
+    private static final Map<Namespace, Long> LOWEST_START_TS_MAP = new HashMap<>();
+    private static final int COMMIT_TS_LIMIT_PER_REQUEST = 5;
+    private static final SafeIllegalStateException EXCEPTION = new SafeIllegalStateException("Something went wrong!");
+    private static final Map<Namespace, LockWatchEventCache> LOCK_WATCH_EVENT_CACHE_MAP_MAP = new HashMap<>();
```

Comment:
private **static** final for this and the `LOWEST_START_TS_MAP` seems a bit odd, this means that `LockWatchEventCache`s will be reused across test methods. Do you need that?